src/Microsoft.ML.TimeSeries/RootCauseLocalizationType.cs

```diff
@@ -0,0 +1,308 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.TimeSeries
+{
+    /// <summary>
+    /// Allows a member to be marked as a <see cref="RootCauseLocalizationInputDataViewType"/>, primarily allowing one to set
+    /// root cause localization input.
+    /// </summary>
+    public sealed class RootCauseLocalizationInputTypeAttribute : DataViewTypeAttribute
+    {
+        /// <summary>
+        /// Create a root cause localizagin input type.
+        /// </summary>
+        public RootCauseLocalizationInputTypeAttribute()
+        {
+        }
+
+        /// <summary>
+        /// Equal function.
+        /// </summary>
+        public override bool Equals(DataViewTypeAttribute other)
+        {
+            if (!(other is RootCauseLocalizationInputTypeAttribute otherAttribute))
+                return false;
+            return true;
+        }
+
+        /// <summary>
+        /// Produce the same hash code for all RootCauseLocalizationInputTypeAttribute.
+        /// </summary>
+        public override int GetHashCode()
+        {
+            return 0;
+        }
+
+        public override void Register()
+        {
+            DataViewTypeManager.Register(new RootCauseLocalizationInputDataViewType(), typeof(RootCauseLocalizationInput), this);
+        }
+    }
+
+    /// <summary>
+    /// Allows a member to be marked as a <see cref="RootCauseDataViewType"/>, primarily allowing one to set
+    /// root cause result.
+    /// </summary>
+    public sealed class RootCauseTypeAttribute : DataViewTypeAttribute
+    {
```

Comment:
Can you please explain why this is necessary?

Reply:
This is the type attribute for root cause output. Root cause output is a customized class(not only double, string such basic type), according to Transformer design, for customized dataview type, a decorated data view type attribute need to be registered. I follow the example of ImageType and when the decorated class is used , the mapping will be registered. Here is the code where ML.net checks the type.
else if (!itemType.TryGetDataKind(out _) && !DataViewTypeManager.Knows(itemType, attributes)) throw Contracts.ExceptParam(nameof(rawType), "Could not determine an IDataView type for member {0}", name);

I have thought of registering the mapping in transformer, but it seems a very hacky way and I haven't found any precedent. If you have better suggestion about the resgitration, feel free to let me know and I will update it.

Reply:
> Can you please explain why this is necessary?

This is the type attribute for root cause output. Root cause output is a customized class(not only double, string such basic type), according to Transformer design, for customized dataview type, a decorated data view type attribute need to be registered. This is the same thing you commented yesterday. One is for input and the other is for output.